mixer/adapter/kubernetes2/template/template.proto

```diff
@@ -0,0 +1,134 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package adapter.template.kubernetes;
+
+import "mixer/v1/template/extensions.proto";
+import "mixer/v1/template/standard_types.proto";
+
+option (istio.mixer.v1.template.template_variety) = TEMPLATE_VARIETY_ATTRIBUTE_GENERATOR;
+
+// kubernetes template represents data used to generate kubernetes attributes.
+//
+//
+// Example config:
+// ```
+// apiVersion: "config.istio.io/v1alpha2"
+// kind: kubernetes
+// metadata:
+//   name: attributes
+//   namespace: istio-system
+// spec:
+//   # Pass the required attribute data to the adapter
+//   source_uid: source.uid | ""
+//   source_ip: source.ip | ip("0.0.0.0") # default to unspecified ip addr
+//   destination_uid: destination.uid | ""
+//   destination_ip: destination.ip | ip("0.0.0.0") # default to unspecified ip addr
+//   attribute_bindings:
+//     # Fill the new attributes from the adapter produced output.
+//     # $out refers to an instance of OutputTemplate message
+//     source.ip: $out.source_pod_ip
+//     source.labels: $out.source_labels
+//     source.namespace: $out.source_namespace
+//     source.service: $out.source_service
+//     source.serviceAccount: $out.source_service_account_name
+//     destination.ip: $out.destination_pod_ip
+//     destination.labels: $out.destination_labels
+//     destination.namespace: $out.destination_mamespace
+//     destination.service: $out.destination_service
+//     destination.serviceAccount: $out.destination_service_account_name
+// ```
+message Template {
+    string source_uid = 1;
```

Comment:
can we add comments here about the expected formats of this data ?

Reply:
D